docs/code-review.md

```diff
@@ -0,0 +1,363 @@
+  *It can affect other services that are not the CLI itself and break scenarios that are not contemplated on the CLI repository, so we need to bear in mind all those services when reviewing a PR*
+- [ ] Does it need to add analytics?
+  *We need to add analytics to know the adoption of new features and how the users use the product*
+- [ ] Does the code have any smell code?
+  we should develop clean code so that if a new developer starts working on the project, he/she can understand the code as soon as possible*
+
+### PR Description
+
+The description of the pull request should state what issue it solves (or explain if there is no associated issue) and how it is solved.
+
+If required, the author could be asked to provide screenshots or videos explaining what was wrong and how it was fixed.
+
+### PR comments
+
+Comments are the basis of all pull requests. In comments, the reviewers will ask questions about the code, request changes, and discuss what is the best approach to improve the CLI code.
+
+A respectful tone will be used when making comments and supporting documentation can be helpful for certain comments.
+
+### Tests
+
+All new features or bug fixes should have unit tests that cover the added code and the functionality it provides. When adding new features it is highly recommended that integration tests are also created and may be deemed required by a reviewer.
+
+Tests should fail with helpful messages saying what was wrong, what inputs were provided, the expected result, and the actual result received in testing. Assume that the person debugging your failing test is not you, and is potentially new to the entire project.
+
+#### Unit tests
+
+The code should be testable by unit functions, which means that each function should perform an action and this action should be testable.
+
+When creating unit tests, try to use table-driven tests when the same result is expected (the function must return an error) for different inputs. Table-driven tests allow us to create a boilerplate in which we define the name of the test (explanation of what it does), the test inputs and the output to be returned. For example:
+
+```golang
+func Test_FunctionNameWithNoError(t *testing.T) {
+    var tests = []struct {
+        name         string
+        resource     Resource
+    }{
+        {
+            name:        "when resource is nil then error",
+            resource:    nil,
+        },
+        {
+            name:        "when resource is not valid then error",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource is valid then no error",
+            resource:    CorrectResource{},
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            assert.NoError(t, err)
+        })
+    }
+}
```

Comment:
I think some part of this example is misleading. The third case expects **no** error according to the description and the other 2 expects **an** error but the test is called `Test_FunctionNameWithNoError`. Also, in the line 92, we assert that **no** error is expected.

I think there is a mix of things between this example and the next one.

I think this first function has to be called `Test_FunctionNameWithError`, remove the third scenario and replace the assert in line 92 by `assert.Error(t, err)`

Then, the second example should be called `Test_FunctionNameWithNoError` (notice the **No**) and the line 100 should be replaced by `err := functionName(CorrectResource{})`